File: code/java/ch07/practice/Warmup04PalindromeNumber.java
package ch07.practice;

import java.util.*;

/**
 * Warmup 04: Palindrome Number
 * ==============================
 * Chapter 7: Number Wizardry
 *
 * PROBLEM: Given integer n, return true if it is a palindrome (reads
 *          the same forwards and backwards). Negative numbers are NOT
 *          palindromes.
 *
 * EXAMPLES:
 *   solve(121)  = true
 *   solve(-121) = false
 *   solve(10)   = false
 *   solve(0)    = true
 *   solve(1001) = true
 *
 * CONSTRAINTS:
 *   -10^18 <= n <= 10^18
 *
 * INSTRUCTIONS: Replace "return false;" in solve() with your solution.
 * Do NOT convert to String!
 */
public class Warmup04PalindromeNumber {
    public static boolean solve(long n) {
        // TODO: Replace this with your solution
        return false;
    }

    // ── Do not change anything below this line ──────────────────────
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        long n = Long.parseLong(sc.nextLine().trim());
        System.out.println(solve(n));
        sc.close();
    }
}

File: code/python/ch07/practice/warmup_04_palindrome_number.py
"""
Warmup 4: Palindrome Number
==============================
Chapter 7: Number Wizardry — Math for Programmers

PROBLEM
-------
Given an integer n, return True if n is a palindrome number, False otherwise.
A palindrome number reads the same forwards and backwards.
Negative numbers are NOT palindromes (e.g., -121 is not a palindrome).

INPUT FORMAT
------------
A single integer n.

OUTPUT FORMAT
-------------
A single value: True or False.

CONSTRAINTS
-----------
- -10^9 <= n <= 10^9

EXAMPLES
--------
Input:
  121
Output: True

Input:
  -121
Output: False

Input:
  10
Output: False

Input:
  0
Output: True

INSTRUCTIONS
------------
Replace the `pass` in the solve() function with your solution.
The main block at the bottom handles input/output — don't change it.
"""


def solve(n: int) -> bool:
    """Return True if n is a palindrome number."""
    pass  # TODO: Replace this with your solution


# ── Do not change anything below this line ──────────────────────────
if __name__ == "__main__":
    n = int(input().strip())
    print(solve(n))
